File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        a = set(nums)
        res = 0
        for n in a:
            if n - 1 not in a:  # sequence 시작점
                length = 1
                while n + length in a:
                    length += 1
                res = max(res, length)
        return res

File: Data Structures & Algorithms/trapping-rain-water/submission-0.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        if (height.empty()) {
            return 0;
        }
        int l = 0;
        int r = height.size() - 1;
        int leftmax = height[l];
        int rightmax = height[r];
        int res = 0;
        
        while (l < r) {
            if (leftmax < rightmax) {
                ++l;
                leftmax = max(height[l], leftmax);
                res += leftmax - height[l];
            }
            else {
                --r;
                rightmax = max(height[r], rightmax);
                res += rightmax - height[r];
            }
        }
        return res;
    }
};
